### PR TITLE
[ntf-core] new port

### PR DESCRIPTION
**Pull request by hallfox**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

Resubmission of #32738 with company CLA attribution

### CODE REVIEW

File: ports/ntf-core/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    SHA512 57662d2dd105b2781e580623c26cd7bde84fce8374bbd70c18595a5f6934869b7a570f0d3c2e17e115f6c7eb1067541f8d19523639815b285324061f807d3179
+    HEAD_REF main
+)
+
+function (ntf_configure var val)
+    set(ENV{NTF_CONFIGURE_${var}} ${val})
+endfunction()
+
+ntf_configure(WITH_MOCKS 0)
+ntf_configure(WITH_APPLICATIONS 0)
```
Comment (FrankXie05) [inline ports/ntf-core/portfile.cmake:14]: All macros can be passed through the `OPTIONS` of function `vcpkg_cmake_configure()`.

File: ports/ntf-core/usage [hunk 2]
```diff
@@ -0,0 +1,5 @@
+The package ntf-core provides the CMake targets:
+
+    find_package(ntc REQUIRED)
+    find_package(nts REQUIRED)
+    target_link_libraries(main PRIVATE ntc nts)
```
Comment (FrankXie05) [inline ports/ntf-core/usage:5]: One port provides two usages? Or two usages one port? 🤔

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". 
          That way, I can be aware that you've responded since you can't modify the tags.

File: ports/ntf-core/usage (hunk 2 above)
Comment (dg0yt) [reply ports/ntf-core/usage:5]: Let me rephrase: Are these independent targets and packages? If not, they need different cmake code (and may need to be fixed for this).

File: ports/ntf-core/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/ntf-core/portfile.cmake:14]: `vcpkg_cmake_configure` doesn't set enironment variables, so it cannot do the same.
But TBH I would prefer to see direct `set(ENV...)` instead of this redirection via `ntf_configure`.

File: ports/ntf-core/usage (hunk 2 above)
Comment (hallfox) [reply ports/ntf-core/usage:5]: Hi, these are coming from the same source project but they are different targets. Perhaps they should be provided as two targets under the same namespace? I'm not sure what the conventions of packages that provide multiple targets are.

Comment (hallfox) [reply ports/ntf-core/usage]: Perhaps to add more clarity regarding independence, both of these targets can be used independently of each other, they are just built from the same source package. The usage text using them both here was more of a shorthand to show that they can be passed as libraries, not that a user must pass both in order to use either.

Comment (dg0yt) [reply ports/ntf-core/usage]: I simply would like to see a few words about what each package/lib does. The names are not very telling. Take a look at `ports/geos/usage` for an example.

File: ports/ntf-core/usage [hunk 3]
```diff
@@ -0,0 +1,4 @@
+The package ntf-core provides the CMake targets:
+
+    find_package(ntc REQUIRED)
+    find_package(nts REQUIRED)
```
Comment (FrankXie05) [inline ports/ntf-core/usage:4]: Reference https://github.com/microsoft/vcpkg/blob/master/ports/geos/usage

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/ntf-core/usage (hunk 2 above)
Comment (hallfox) [reply ports/ntf-core/usage]: Ok, will do!

**Comment by hallfox** [comment]
@FrankXie05 everything look good here?

**Review by FrankXie05 — DISMISSED** [review]

**Comment by FrankXie05** [comment]
@hallfox  Test error on `x64-linux` locally:
```
Installing 4/4 ntf-core:x64-linux...
Building ntf-core:x64-linux...
-- Downloading https://github.com/bloomberg/ntf-core/archive/2.1.0.tar.gz -> bloomberg-ntf-core-2.1.0.tar.gz...
-- Extracting source /home/test/vcpkg/downloads/bloomberg-ntf-core-2.1.0.tar.gz
-- Using source at /home/test/vcpkg/buildtrees/ntf-core/src/2.1.0-b5bd59a649.clean
-- Configuring x64-linux
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /home/test/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja -v
    Working Directory: /home/test/vcpkg/buildtrees/ntf-core/x64-linux-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      /home/test/vcpkg/buildtrees/ntf-core/config-x64-linux-dbg-CMakeCache.txt.log
      /home/test/vcpkg/buildtrees/ntf-core/config-x64-linux-rel-CMakeCache.txt.log
      /home/test/vcpkg/buildtrees/ntf-core/config-x64-linux-out.log

 

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/ntf-core/portfile.cmake:9 (vcpkg_cmake_configure)
  scripts/ports.cmake:147 (include)
```

`/home/test/vcpkg/buildtrees/ntf-core/config-x64-linux-out.log`:
```
CMake Error at repository.cmake:3916 (find_program):
  Could not find debugger_path using the following names: gdb
Call Stack (most recent call first):
  targets.cmake:18 (ntf_repository)
  CMakeLists.txt:42 (include)
```

**Comment by hallfox** [comment]
Argh, the current version of ntf-core seems to require a debugger on configuring. I'm working with the maintainer to remove this requirement. Is there a way to require a debugger in the dependencies, or are we just stuck until this requirement is removed?

**Comment by dg0yt** [comment]
> Is there a way to require a debugger in the dependencies

No.

**Comment by hallfox** [comment]
Sounds good, I've updated the config to use a fake variable for the debugger path, this shouldn't affect the build except to let it continue without a debugger installed. Future releases of ntf-core will remove the requirement for the debugger, so I expect we can remove this option in the future.

**Comment by hallfox** [comment]
Checking in, is this good to go?

**Review by dan-shaw — APPROVED** [review]
